### ISSUE TITLE
With AWS Secrets Manager backend, make one AWS API call per secret even when not using dataFrom

### ISSUE THREAD

**Issue by mbonica1990** [external-secrets/external-secrets#114]
(I had initially posted this the Kubernetes External Secrets project as https://github.com/external-secrets/kubernetes-external-secrets/issues/704 , but it was pointed out to me that it might more sense to talk about this here since this is where the project's moving.)

I'm using secretsManager as my backend for some k8s secrets; my ExternalSecret object looks a bit like this.

```yaml
apiVersion: kubernetes-client.io/v1
kind: ExternalSecret
metadata:
  name: my-external-secret
spec:
  backendType: secretsManager
  data:
    - key: /path/to/secret
      name: username
      property: username
    - key: /path/to/secret
      name: password
      property: password`
```
      
  Notice that I have one Secrets Manager secret, called `/path/to/secret`, that I'm getting two values from (`username` and `password`) and surfacing as two different pieces of data in my secret.
  
 This works, but I can see through CloudTrail that the External Secrets pod makes `two` API requests to Secrets Manager (one per item in the `data` list). I'm wondering if there's a way to get I can make it get both values in just `one` API call, since I imagine the response it's getting from the first call also has the values needed for the second. I'd like to minimize the number of API calls I make to AWS to avoid getting throttled.
 
 I understand that `dataFrom` would solve my problem, but I'm hoping to avoid using that if I can; we'd like to be explicit about what secrets we're including in our applications, and `dataFrom` doesn't make it clear when looking at the YAML which values I'm bringing into my application. I'd be just as happy with an option to explicitly call out which keys I'm getting from `dataFrom`.
 
```yaml
apiVersion: kubernetes-client.io/v1
kind: ExternalSecret
metadata:
  name: my-external-secret
spec:
  backendType: secretsManager
  dataFromSubset:
    secret: /path/to/secret
    keys:
      - username
      - password
```

Where "dataFromSubset" acted just like "dataFrom", but only surfaced the `keys` specifically called out here.

**Comment by moolen** [comment]
Thank you for that detailed write-up! I think it makes sense to optimize the fetching of the secret. From a first look i'd propose to cache the fetched secret value in the aws secretsmanager client ([here](https://github.com/external-secrets/external-secrets/blob/main/pkg/provider/aws/secretsmanager/secretsmanager.go#L30-L32)). The provider client is constructed in the reconciliation loop ([here](https://github.com/external-secrets/external-secrets/blob/main/pkg/controllers/externalsecret/externalsecret_controller.go#L93)) and the cache is discarded after the secret was reconciled.

BTW, the custom resource schema looks different here ([from docs](https://external-secrets.io/api-overview/#externalsecret)):
```yaml
apiVersion: external-secrets.io/v1alpha1
kind: ExternalSecret
metadata:
  name: example
spec:
  secretStoreRef:
    name: secretstore-sample
    kind: SecretStore
  target:
    name: my-api-credentials
  data:
  - secretKey: my-user
    remoteRef:
      key: /path/to/secret
      property: username
  - secretKey: my-pass
    remoteRef:
      key: /path/to/secret
      property: password

```

**Comment by alekc** [comment]
If nobody is working on this one I can take it on following @moolen solution (which sounds reasonable to me)?

**Comment by moolen** [comment]
fixed with #334
